airflow/dag_processing/processor.py

```diff
@@ -689,19 +691,18 @@ def _execute_task_callbacks(self, dagbag: DagBag | None, request: TaskCallbackRe
             return
 
         simple_ti = request.simple_task_instance
-        ti: TI | None = (
-            session.query(TI)
-            .filter_by(
-                dag_id=simple_ti.dag_id,
-                run_id=simple_ti.run_id,
-                task_id=simple_ti.task_id,
-                map_index=simple_ti.map_index,
-            )
-            .one_or_none()
+        ti = InternalApiTaskInstanceActions.get_task_instance(
```

Comment:
Alternativele you can `get_task_instance` method in this file (with `@internal_ap_call` decorator) so the whole logic is in a single place.